src/dbnode/storage/series/buffer.go

```diff
@@ -951,7 +962,9 @@ func (b *BufferBucket) writeToEncoderIndex(
 	datapoint ts.Datapoint,
 	unit xtime.Unit,
 	annotation []byte,
+	schema namespace.SchemaDescr,
 ) error {
+	b.encoders[idx].encoder.SetSchema(schema)
```

Comment:
Hm, this is going to be setting the schema each and every write.  Is a SetSchema for the same schema a no-op?  If not that may be an issue.

Reply:
we compare version (string compare), if same, no-op.

see SetSchema (or resetSchema) in proto/encode|iterator.go

Reply:
Ok no problem, this is fine.  If it's possible to just make this part of Reset(...) like the other args (such as namespace context were moved to Reset) that'd be great, if it's too difficult then let's just punt on this for now.